Tests/SwiftFormatPrettyPrintTests/TernaryExprTests.swift

```diff
@@ -52,4 +55,170 @@ public class TernaryExprTests: PrettyPrintTestCase {
 
     assertPrettyPrintEqual(input: input, expected: expected, linelength: 80)
   }
+
+  public func testTernaryWithWrappingExpressions() {
+    let input =
+      """
+      foo = firstTerm + secondTerm + thirdTerm ? firstTerm + secondTerm + thirdTerm : firstTerm + secondTerm + thirdTerm
+      let foo = firstTerm + secondTerm + thirdTerm ? firstTerm + secondTerm + thirdTerm : firstTerm + secondTerm + thirdTerm
+      foo = firstTerm || secondTerm && thirdTerm ? firstTerm + secondTerm + thirdTerm : firstTerm + secondTerm + thirdTerm
+      let foo = firstTerm || secondTerm && thirdTerm ? firstTerm + secondTerm + thirdTerm : firstTerm + secondTerm + thirdTerm
+      """
+
+    let expected =
+      """
+      foo =
+        firstTerm
+          + secondTerm
+          + thirdTerm
+        ? firstTerm
+          + secondTerm
+          + thirdTerm
+        : firstTerm
+          + secondTerm
+          + thirdTerm
+      let foo =
+        firstTerm
+          + secondTerm
+          + thirdTerm
+        ? firstTerm
+          + secondTerm
+          + thirdTerm
+        : firstTerm
+          + secondTerm
+          + thirdTerm
+      foo =
+        firstTerm
+          || secondTerm
+            && thirdTerm
+        ? firstTerm
+          + secondTerm
+          + thirdTerm
+        : firstTerm
+          + secondTerm
+          + thirdTerm
+      let foo =
+        firstTerm
+          || secondTerm
+            && thirdTerm
+        ? firstTerm
+          + secondTerm
+          + thirdTerm
+        : firstTerm
+          + secondTerm
+          + thirdTerm
+
+      """
+
+    assertPrettyPrintEqual(input: input, expected: expected, linelength: 15)
+  }
+
+  public func testNestedTernaries() {
+    let input =
+      """
+      a = b ? c : d ? e : f
+      let a = b ? c : d ? e : f
+      a = b ? c0 + c1 : d ? e : f
+      let a = b ? c0 + c1 : d ? e : f
+      a = b ? c0 + c1 + c2 + c3 : d ? e : f
+      let a = b ? c0 + c1 + c2 + c3 : d ? e : f
+      foo =
+        testA
+        ? testB
+          ? testC
+          : testD
+        : testE
+          ? testF
+          : testG
+      let foo =
+        testA
+        ? testB
+          ? testC
+          : testD
+        : testE
+          ? testF
+          : testG
+      """
+
+    let expected =
+      """
+      a =
+        b
+        ? c
+        : d ? e : f
+      let a =
+        b
+        ? c
+        : d ? e : f
+      a =
+        b
+        ? c0 + c1
+        : d ? e : f
+      let a =
+        b
+        ? c0 + c1
+        : d ? e : f
+      a =
+        b
+        ? c0 + c1
+          + c2 + c3
+        : d ? e : f
+      let a =
+        b
+        ? c0 + c1
+          + c2 + c3
+        : d ? e : f
+      foo =
```

Comment:
Woops, not only would that be interesting to check, it's what I intended to check. I wrote the input string with indentation so that I wouldn't screw up the pairing of the operators. Fixed!